src/saltext/kubernetes/states/kubernetes.py

```diff
@@ -0,0 +1,1488 @@
+
+        log.info("Forcing the recreation of the service")
+        ret["comment"] = "The configmap is already present. Forcing recreation"
+        try:
+            res = __salt__["kubernetes.replace_configmap"](
+                name=name,
+                namespace=namespace,
+                data=data,
+                source=source,
+                template=template,
+                saltenv=__env__,
+                template_context=template_context,
+                wait=wait,
+                timeout=timeout,
+                **kwargs,
+            )
+        except CommandExecutionError as err:
+            log.exception(str(err), exc_info_on_loglevel=logging.DEBUG)
+            ret["result"] = False
+            ret["comment"] = str(err)
+            return ret
+
+    ret["changes"] = {"data": res["data"]}
+    ret["result"] = True
+    return ret
+
+
+def pod_absent(name, namespace="default", wait=False, timeout=60, **kwargs):
+    """
+    Ensures that the named pod is absent from the given namespace.
+
+    name
+        The name of the pod
+
+    namespace
+        The name of the namespace
+
+    wait
+        .. versionadded:: 2.0.0
+
+        If set to True, the function will wait until the pod is deleted.
+
+    timeout
+        .. versionadded:: 2.0.0
+
+        The time in seconds to wait for the pod to be deleted.
+
+    Example:
+
+    .. code-block:: yaml
+
+        my_pod:
+          kubernetes.pod_absent:
+            - namespace: default
+    """
+    ret = {"name": name, "changes": {}, "result": False, "comment": ""}
+
+    try:
+        pod = __salt__["kubernetes.show_pod"](name, namespace, **kwargs)
+    except CommandExecutionError as err:
+        log.exception(str(err), exc_info_on_loglevel=logging.DEBUG)
+        ret["result"] = False
+        ret["comment"] = str(err)
+        return ret
+
+    if pod is None:
+        ret["result"] = True if not __opts__["test"] else None
+        ret["comment"] = "The pod does not exist"
+        return ret
+
+    if __opts__["test"]:
+        ret["comment"] = "The pod is going to be deleted"
+        ret["result"] = None
+        return ret
+
+    try:
+        __salt__["kubernetes.delete_pod"](name, namespace, wait=wait, timeout=timeout, **kwargs)
+    except CommandExecutionError as err:
+        log.exception(str(err), exc_info_on_loglevel=logging.DEBUG)
+        ret["result"] = False
+        ret["comment"] = str(err)
+        return ret
+
+    ret["result"] = True
+    ret["changes"] = {"kubernetes.pod": {"new": "absent", "old": "present"}}
+    ret["comment"] = "Pod deleted"
+    return ret
+
+
+def pod_present(
+    name,
+    namespace="default",
+    metadata=None,
+    spec=None,
+    source="",
+    template="",
+    template_context=None,
+    wait=False,
+    timeout=60,
+    **kwargs,
+):
+    """
+    Ensures that the named pod is present inside of the specified
+    namespace with the given metadata and spec.
+    If the pod exists it will be replaced.
```

Comment:
issue: This statement is false - the state currently fails if the pod exists.

I think that's bad logic, but it should be fixed in another PR.

suggestion for now: Correct the statement, similar to what the error message says.

suggestion for some other time: Introduce a `replace` parameter, which deletes the pod before recreating it. Otherwise, return `ret["result"] = True` and add a note in `ret["comment"]` that the state is not idempotent.